plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestReusableConnectionFactory.java

```diff
@@ -139,6 +139,28 @@ public void testConnectionIsNotReusedWhenSetToReadOnly()
         }
     }
 
+    @Test
+    public void testConnectionIsNotReusedWhenDelegateIsClosed()
+            throws Exception
+    {
+        MockConnectionFactory mockConnectionFactory = new MockConnectionFactory();
+        ReusableConnectionFactory connectionFactory = new ReusableConnectionFactory(mockConnectionFactory, FOREVER, HUGE_SIZE);
+        Connection connection = connectionFactory.openConnection(ALICE);
+
+        Connection delegate = ((ReusableConnectionFactory.CachedConnection) connection).delegate();
+        delegate.close();
+        connection.close();
+
+        Connection secondConnection = connectionFactory.openConnection(ALICE);
+        Connection secondDelegate = ((ReusableConnectionFactory.CachedConnection) secondConnection).delegate();
+
+        assertThat(delegate).isNotEqualTo(secondDelegate);
```

Comment:
I'd love also to see positive assertion 
`assertThat(delegate).isEqualTo(firstDelegate);`
or
`assertThat(delegate).isEqualTo(thirdDelegate);`

Reply:
It's covered by other tests

Reply:
agree. But it would make it easier to compare what has changed.